### PR TITLE
[SPARK-41566][BUILD] Upgrade `netty`  to 4.1.86.Final

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Upgrade netty from 4.1.84.Final to 4.1.86.Final

[Release notes for .86](https://netty.io/news/2022/12/12/4-1-86-Final.html)

[Release notes for .85](https://netty.io/news/2022/11/09/4-1-85-Final.html)
### Why are the changes needed?

To fix 2 CVE's
[HAProxyMessageDecoder Stack Exhaustion DoS](https://github.com/netty/netty/security/advisories/GHSA-fx2c-96vj-985v)

and 

[HTTP Response splitting from assigning header value iterator](https://github.com/netty/netty/security/advisories/GHSA-hh82-3pmq-7frp)

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

File: dev/deps/spark-deps-hadoop-2-hive-2.3 [hunk 1]
```diff
@@ -200,24 +200,25 @@ metrics-jmx/4.2.13//metrics-jmx-4.2.13.jar
...
-netty-transport/4.1.84.Final//netty-transport-4.1.84.Final.jar
+netty-all/4.1.86.Final//netty-all-4.1.86.Final.jar
+netty-buffer/4.1.86.Final//netty-buffer-4.1.86.Final.jar
+netty-codec-http/4.1.86.Final//netty-codec-http-4.1.86.Final.jar
+netty-codec-http2/4.1.86.Final//netty-codec-http2-4.1.86.Final.jar
+netty-codec-socks/4.1.86.Final//netty-codec-socks-4.1.86.Final.jar
+netty-codec/4.1.86.Final//netty-codec-4.1.86.Final.jar
+netty-common/4.1.86.Final//netty-common-4.1.86.Final.jar
+netty-handler-proxy/4.1.86.Final//netty-handler-proxy-4.1.86.Final.jar
+netty-handler-ssl-ocsp/4.1.86.Final//netty-handler-ssl-ocsp-4.1.86.Final.jar
```
Comment (LuciferYang) [inline dev/deps/spark-deps-hadoop-2-hive-2.3:211]: `netty-handler-ssl-ocsp`, seems a new dependency. Does Spark really need it?

Comment (bjornjorgensen) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: I don't think so, but maven did add this when I run `./dev/test-dependencies.sh --replace-manifest`

Comment (LuciferYang) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: Is it easy to manually exclude?

Comment (bjornjorgensen) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: I don't know. I don't think we should do it. 
Let it be there.

Comment (LuciferYang) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: If we are sure that we do not need this new jar, I suggest remove it, WDYT @HyukjinKwon

Comment (bjornjorgensen) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: https://github.com/netty/netty/issues/13003 they have done this on purpose

Comment (bjornjorgensen) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:211]: just to add it her 
https://github.com/apache/spark/blob/ae5c9eb6dd241e7916101c302f1fa84a8fe08141/pom.xml#L894

Comment (bjornjorgensen) [reply dev/deps/spark-deps-hadoop-2-hive-2.3]: @LuciferYang Thank you :) 
I have now added netty-handler-ssl-ocsp to exclude list. and run `./dev/test-dependencies.sh --replace-manifest`

**Review by LuciferYang — APPROVED** [review]
+1, LGTM

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**